src/bid/encoding.rs

```diff
@@ -5,12 +5,48 @@
 use super::Bid;
 use dusk_bls12_381::Scalar;
 use dusk_plonk::constraint_system::{StandardComposer, Variable};
+use jubjub::{AffinePoint as JubJubAffine, Scalar as JubJubScalar};
 use poseidon252::{sponge::sponge::*, StorageScalar};
 
-/// Encodes a `Bid` in a `StorageScalar` form by applying the correct encoding methods
+#[derive(Copy, Clone, Debug, Default)]
+pub struct StorageBid {
+    // t_a
+    pub(crate) elegibility_ts: u32,
+    // t_e
+    pub(crate) expiration_ts: u32,
+    // b_enc (encrypted blinder) // XXX: Scalar for now. Double check
+    pub(crate) encrypted_blinder: JubJubScalar,
+    // v_enc (encrypted_value)
+    pub(crate) encrypted_value: JubJubScalar,
+    // R = r * G
+    pub(crate) randomness: JubJubAffine,
+    // m
+    pub(crate) hashed_secret: Scalar,
+    // pk (Public Key - Stealth Address)
+    pub(crate) pk: JubJubAffine,
+    // c (Pedersen Commitment)
+    pub(crate) c: JubJubAffine,
+}
+
+impl From<&Bid> for StorageBid {
```

Comment:
It's fine for now, but take in account that this code most likely will be moved / refactored from here because it doesn't strictly belong here, and it might need be a structure that a contract can understand (so my rough idea is that it will be part of the blind bid ABI)

Reply:
Fair enough.

Once I start coding it we'll be able to tackle this more precisely!